source/client/stream_decoder.cc

```diff
@@ -19,6 +19,19 @@ void StreamDecoder::decodeHeaders(Envoy::Http::ResponseHeaderMapPtr&& headers, b
   response_header_sizes_statistic_.addValue(response_headers_->byteSize());
   const uint64_t response_code = Envoy::Http::Utility::getResponseStatus(*response_headers_);
   stream_info_.response_code_ = static_cast<uint32_t>(response_code);
+  const auto timing_header_name = Envoy::Http::LowerCaseString("x-nh-do-not-use-origin-timings");
+  const auto* timing_header = response_headers_->get(timing_header_name);
```

Comment:
(nit) Since the type isn't obvious from the statement, it may be a good idea to replace auto with the actual type here to improve readability.

Applies to timing_value below as well.